File: backend/backend/Models/Subject.cs
using Microsoft.AspNetCore.Identity;
using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
using Newtonsoft.Json;
using Swashbuckle.AspNetCore.Annotations;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    public class Subject
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public string Id { get; set; }
        public string? Name { get; set; }
        public string? Neptun { get; set; }
        public int? Credit { get; set; }
        public bool? Exam { get; set; }
        public string? Image { get; set; }

        [NotMapped]
        public virtual ICollection<Teacher>? Teachers { get; set; }

        [NotMapped]
        [JsonIgnore]
        public virtual Student? Student { get; set; }
        public string? StudentId { get; set; }

        public Subject()
        {
            Teachers = new List<Teacher>();
        }
    }
}

File: frontend/src/app/_models/subject.ts
import { Teacher } from "./teacher"

export class Subject {
    public id:  string = ''
    public name: string = ''
    public neptun: string = ''
    public credit: number = 0
    public exam: boolean = false
    public image: string = ''
    public Teachers: Array<Teacher> = []

    public createTeachers(teacherList: Array<any>) {
        if (teacherList) {
            teacherList.map((x:any) => {
                let teacher = new Teacher()
                teacher.id = x.id
                teacher.name = x.name
                teacher.neptun = x.neptun
                teacher.image = x.image
                this.Teachers.push(teacher)
            })
        }
    }   
}